File: java-vertx/src/main/java/org/sample/MainVerticle.java
package org.sample;

import io.vertx.core.AbstractVerticle;
import io.vertx.ext.web.Router;

/**
 * @author dyingleaf3213
 * @Classname VertxApp
 * @Description TODO
 * @Create 2023/03/27 21:24
 */

public class MainVerticle extends AbstractVerticle {

    @Override
    public void start() throws Exception {

        // Create a Router
        Router router = Router.router(vertx);

        //
        //router = AppRouterBuilder.build(router);

        // Create the HTTP server
        vertx.createHttpServer()
                // Handle every request using the router
                .requestHandler(router)
                // Start listening
                .listen(8888)
                // Print the port
                .onSuccess(server -> {
                    System.out.println("HTTP server started on port " + server.actualPort());
                });
    }
}

File: vertx-runner/src/main/kotlin/code/sibyl/MainVerticle.kt
package code.sibyl

import io.vertx.core.AbstractVerticle
import io.vertx.core.Promise
import io.vertx.ext.web.Router
import io.vertx.ext.web.RoutingContext
import io.vertx.ext.web.common.template.TemplateEngine
import io.vertx.ext.web.handler.StaticHandler
import io.vertx.ext.web.handler.TemplateHandler
import io.vertx.ext.web.templ.thymeleaf.ThymeleafTemplateEngine


class MainVerticle : AbstractVerticle() {

    override fun start(startPromise: Promise<Void>) {

        val engine: TemplateEngine? = ThymeleafTemplateEngine.create(vertx)
        val handler: TemplateHandler = TemplateHandler.create(engine)

        val router: Router = Router.router(vertx)
        router.route("/dist/*").handler(StaticHandler.create("static/dist"));
        router.route("/css/*").handler(StaticHandler.create("static/css"));
        router.route("/static/*").handler(StaticHandler.create("static/static"));

        router.get("/").handler(handler);
        router.getWithRegex(".+\\.html").handler(handler);

        router["/"]
                .handler { ctx: RoutingContext -> ctx.reroute("/index.html") }
        router["/index"]
                .handler { ctx: RoutingContext -> ctx.reroute("/index.html") }

        vertx
                .createHttpServer()
                .requestHandler(router)
                .listen(8888) { http ->
                    if (http.succeeded()) {
                        startPromise.complete()
                        println("HTTP server started on port 8888")
                    } else {
                        startPromise.fail(http.cause());
                    }
                }
    }
}